apis/duck/v1/podspec_types.go

```diff
@@ -44,7 +45,8 @@ type WithPod struct {
 
 // WithPodSpec is the shell around the PodSpecable within WithPod.
 type WithPodSpec struct {
-	Template PodSpecable `json:"template,omitempty"`
+	Selector *metav1.LabelSelector `json:"selector"`
```

Comment:
This would break compatibility with our own PodSpecables, like Service and Configuration.

Reply:
If that's the case then we need to write new one because for controllers selector is mandatory

Reply:
See my comment on the PR. I think we shouldn't actually reserialize from the PodSpecable as that's a lossy conversion (as you noticed).

Instead, we should only override the relevant part of the unstructured that's been updated and part of PodSpecable